src/main/scala/firrtl/backends/experimental/smt/Btor2Serializer.scala

```diff
@@ -48,15 +48,15 @@ private class Btor2Serializer private () {
   }
 
   // bit vector type serialization
-  private val bitVecTypeCache = mutable.HashMap[Int, Int]()
+  private val bitVecTypeCache = mutable.HashMap[BigInt, Int]()
 
-  private def t(width: Int): Int = bitVecTypeCache.getOrElseUpdate(width, line(s"sort bitvec $width"))
+  private def t(width: BigInt): Int = bitVecTypeCache.getOrElseUpdate(width, line(s"sort bitvec $width"))
 
   // bit vector expression serialization
   private def s(expr: BVExpr): Int = expr match {
     case BVLiteral(value, width) => lit(value, width)
     case BVSymbol(name, _)       => symbols(name)
-    case BVExtend(e, 0, _)       => s(e)
+    case BVExtend(e, by, _) if by == 0 => s(e)
```

Comment:
This is why I did not want to use BigInt: They do not work well for pattern matching.

Reply:
No worries, that's fine - will put it back to Int. I don't have a strong opinion either way. Was just noting that for some reason the rest of FIRRTL uses BigInt for widths.